src/sentry/integrations/msteams/card_builder/__init__.py

```diff
@@ -22,348 +18,20 @@
 ColumnBlock = Mapping[str, Union[str, Sequence[ItemBlock]]]
 ColumnSetBlock = Mapping[str, Union[str, Sequence[ColumnBlock]]]
 
-Block = Union[TextBlock, ImageBlock, ColumnSetBlock]
+# NOTE: Instead of Any, it should have been block, but mypy does not support cyclic definition.
+ContainerBlock = Mapping[str, Any]
+
+Block = Union[TextBlock, ImageBlock, ColumnSetBlock, ContainerBlock]
+
+InputChoiceSetBlock = Mapping[str, Union[str, Sequence[Mapping[str, Any]]]]
 
 # Maps to Any because Actions can have an arbitrarily nested data field.
 Action = Mapping[str, Any]
+ActionSet = Mapping[str, Union[str, Sequence[Action]]]
 
 AdaptiveCard = Mapping[str, Union[str, Sequence[Block], Sequence[Action]]]
 
 
-def generate_action_payload(action_type, event, rules, integration):
-    # we need nested data or else Teams won't handle the payload correctly
-    return {
-        "payload": {
-            "actionType": action_type,
-            "groupId": event.group.id,
-            "eventId": event.event_id,
-            "rules": [rule.id for rule in rules],
-            "integrationId": integration.id,
-        }
-    }
-
-
-def get_assignee_string(group: Group) -> str | None:
-    """Get a string representation of the group's assignee."""
-    assignee = group.get_assignee()
-    if isinstance(assignee, User):
-        return assignee.email
-
-    if isinstance(assignee, Team):
-        return f"#{assignee.slug}"
-
-    return None
-
-
-def build_group_title(group):
```

Comment:
Already implemented in `build_attachment_title` and `get_title_link` in `slack.message_builder.issues`.